backend/remote-state/gcs/backend.go

```diff
@@ -45,13 +45,6 @@ func New() backend.Backend {
 				Description: "The name of the Google Cloud Storage bucket",
 			},
 
-			"path": {
```

Comment:
Watch out for line 117, where this is still in use: `data.Get("path")`

It appears the `defaultStateFile` value is also no longer relevant, since that was the sole thing that set it.

Reply:
`remove deadcode` removed the last non-test setter, but I think we can remove it at https://github.com/hashicorp/terraform/blob/master/backend/remote-state/gcs/backend.go#L32 and in https://github.com/hashicorp/terraform/blob/master/backend/remote-state/gcs/backend_state.go + https://github.com/hashicorp/terraform/blob/master/backend/remote-state/gcs/backend_test.go too?

Reply:
Trying it now, getting some fun errors when i remove defaultStateFile from the struct and clean up references to it

Reply:
It is working now, not sure what the context for the legacy test is for.

```
REDACTED  MCW0CDP3YY  ~  Desktop  Git  terraform   remove-path-gcs-be  3✎  ERROR  $   TF_ACC=1 go test -v ./backend/remote-state/gcs/
=== RUN   TestStateFile
=== PAUSE TestStateFile
=== RUN   TestRemoteClient
=== PAUSE TestRemoteClient
=== RUN   TestRemoteClientWithEncryption
=== PAUSE TestRemoteClientWithEncryption
=== RUN   TestRemoteLocks
=== PAUSE TestRemoteLocks
=== RUN   TestBackend
=== PAUSE TestBackend
=== RUN   TestBackendWithPrefix
=== PAUSE TestBackendWithPrefix
=== RUN   TestBackendWithEncryption
=== PAUSE TestBackendWithEncryption
=== CONT  TestStateFile
=== CONT  TestBackend
--- PASS: TestStateFile (0.00s)
=== CONT  TestRemoteClient
=== CONT  TestBackendWithEncryption
=== CONT  TestBackendWithPrefix
=== CONT  TestRemoteLocks
    TestRemoteLocks: backend_test.go:96: TestBackendConfig on *gcs.Backend with configs.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("tf-1645030c4adee3b0-testremotelocks"), "encryption_key":cty.StringVal(""), "prefix":cty.StringVal("")}}
=== CONT  TestRemoteClientWithEncryption
    TestBackendWithPrefix: backend_test.go:146: TestBackendConfig on *gcs.Backend with configs.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("tf-1645030c4ade0508-testbackendwithprefix"), "encryption_key":cty.StringVal(""), "prefix":cty.StringVal("test/prefix")}}
    TestBackendWithEncryption: backend_test.go:159: TestBackendConfig on *gcs.Backend with configs.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("tf-1645030c4addd628-testbackendwithencryption"), "encryption_key":cty.StringVal("yRyCOikXi1ZDNE0xN3yiFsJjg7LGimoLrGFcLZgQoVk="), "prefix":cty.StringVal("")}}
    TestBackend: backend_test.go:130: TestBackendConfig on *gcs.Backend with configs.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("tf-1645030c4adc14f0-testbackend"), "encryption_key":cty.StringVal(""), "prefix":cty.StringVal("")}}
    TestRemoteClient: backend_test.go:57: TestBackendConfig on *gcs.Backend with configs.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("tf-1645030c4add39e8-testremoteclient"), "encryption_key":cty.StringVal(""), "prefix":cty.StringVal("")}}
    TestRemoteClientWithEncryption: backend_test.go:76: TestBackendConfig on *gcs.Backend with configs.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("tf-1645030c4ae5cd38-testremoteclientwithencryption"), "encryption_key":cty.StringVal("yRyCOikXi1ZDNE0xN3yiFsJjg7LGimoLrGFcLZgQoVk="), "prefix":cty.StringVal("")}}
    TestBackendWithPrefix: backend_test.go:149: TestBackendConfig on *gcs.Backend with configs.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("tf-1645030c4ade0508-testbackendwithprefix"), "encryption_key":cty.StringVal(""), "prefix":cty.StringVal("test/prefix/")}}
    TestBackendWithEncryption: backend_test.go:162: TestBackendConfig on *gcs.Backend with configs.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("tf-1645030c4addd628-testbackendwithencryption"), "encryption_key":cty.StringVal("yRyCOikXi1ZDNE0xN3yiFsJjg7LGimoLrGFcLZgQoVk="), "prefix":cty.StringVal("")}}
    TestBackend: backend_test.go:133: TestBackendConfig on *gcs.Backend with configs.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("tf-1645030c4adc14f0-testbackend"), "encryption_key":cty.StringVal(""), "prefix":cty.StringVal("")}}
--- PASS: TestRemoteClient (5.46s)
--- PASS: TestRemoteClientWithEncryption (5.79s)
--- PASS: TestRemoteLocks (6.37s)
    TestBackendWithPrefix: backend_test.go:152: TestBackend: testing state locking for *gcs.Backend
    TestBackend: backend_test.go:136: TestBackend: testing state locking for *gcs.Backend
    TestBackendWithEncryption: backend_test.go:165: TestBackend: testing state locking for *gcs.Backend
--- PASS: TestBackendWithPrefix (15.32s)
    TestBackend: backend_test.go:137: TestBackend: testing state locking for *gcs.Backend
--- PASS: TestBackendWithEncryption (15.87s)
--- PASS: TestBackend (19.85s)
PASS
ok      github.com/hashicorp/terraform/backend/remote-state/gcs 22.278s
```